### ISSUE TITLE
Bug: Mermaid diagrams not working on lesson previews

### ISSUE THREAD

**Issue by KevinMulhern** [TheOdinProject/theodinproject#4783]
### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] The title of this issue follows the `Bug: brief description of bug` format, e.g. `Bug: Lesson complete button does not update on click`
- [ ] Would you like to work on this issue?

### Bug description

Mermaid diagrams aren't rendering correctly on our lesson preview tool.

### How to reproduce

1. Visit /lessons/previews
2. Paste the contents of the [Revisiting Rock Paper Scissors](https://github.com/TheOdinProject/curriculum/edit/main/foundations/javascript_basics/revisiting_rock_paper_scissors.md) lesson into the input
3. Click the preview button and scroll down to the "Using branches section"
4. 💥 the diagram will be broken and displayed like this:
<img width="669" alt="Screenshot 2024-08-23 at 09 20 18" src="https://github.com/user-attachments/assets/98018d39-63b2-4da0-b8e2-1d7ac9a5672b">

### Expected behavior

1. Visit /lessons/previews
2. Paste the contents of the [Revisiting Rock Paper Scissors](https://github.com/TheOdinProject/curriculum/edit/main/foundations/javascript_basics/revisiting_rock_paper_scissors.md) lesson into the input
3. Click the preview button and scroll down to the "Using branches section"
4. The diagram will be displayed like this:
<img width="741" alt="Screenshot 2024-08-23 at 09 21 24" src="https://github.com/user-attachments/assets/910ed02c-b085-46c6-880b-01f0fef0a378">

### What browsers are you seeing the problem on?

Chrome

### What OS are you using?

MacOS

### Discord Name

KM

### Additional Comments

The markup appears to be correct, I think it might be an issue with how we're initializing mermaid when the diagramming Stimulus controller connects.

**Comment by MaoShizhong** [comment]
Haven't had a huge amount of time to explore this, but it's likely due to the `hidden` class on the container for the markdown preview itself (where the `hidden` class is added on load due to `app/javascript/controllers/tabs_controller.js`).

If you comment out the contents of `TabsController.prototype.showTab` so it doesn't get the hidden class on load, then throw in the mermaid code, it'll render just fine.

If you manually add `hidden` to the div in `app/views/lessons/previews/show.html.erb` and do the same, when you remove the `hidden` class in devtools then the diagram is incorrectly rendered.

So basically, when mermaid tries to render the diagram (the moment it appears in the textarea), if the diagram itself is inside a `hidden` container, it gets all the wrong SVG values.

**Comment by MaoShizhong** [comment]
Got a simple fix!
